cropduster/views/__init__.py

```diff
@@ -419,13 +433,24 @@ def crop(request):
         elif thumb.pk and thumb.name and thumb.crop_w and thumb.crop_h:
             thumb_path = db_image.get_image_path(thumb.name, tmp=False)
             tmp_thumb_path = db_image.get_image_path(thumb.name, tmp=True)
-            if os.path.exists(thumb_path):
-                if not thumb_form.cleaned_data.get('changed') or not os.path.exists(tmp_thumb_path):
-                    shutil.copy(thumb_path, tmp_thumb_path)
+
+            if default_storage.exists(thumb_path):
+                if not thumb_form.cleaned_data.get('changed') or not default_storage.exists(tmp_thumb_path):
+                    file = default_storage.open(thumb_path)
+                    tmp_file = default_storage.open(tmp_thumb_path, 'w')
+                    image = BytesIO(file.read())
+                    tmp_image = BytesIO()
+                    file.close()
+                    img = PIL.Image.open(image)
+                    img.save(tmp_image, format=tmp_thumb_path.split('.')[-1])
```

Comment:
This needs to be:

```python
img.save(tmp_image, format=img.format)
```

There is a similar line in utils that's causing the tests to fail (because the format name is JPEG, not JPG).

Reply:
A-ha, cheers 👍